Some PVS-Studio fixes

dimhotepus opened the pull request:
Product page: https://www.viva64.com/en/pvs-studio/
How to use for free on open source projects: https://www.viva64.com/en/b/0600/

arves100 commented:
Hello, thanks for your PR, I'll review it in chunks during this days, just to know, you runned the latest version of PVS Studio right? (I never used that program)

dimhotepus commented:
> Hello, thanks for your PR, I'll review it in chunks during this days, just to know, you runned the latest version of PVS Studio right? (I never used that program)

Yep, the latest one. Unfortunately i can't check the solution builds, as some functions are missed (Sha1*, MD5*, etc.) May be GS* equivalents should be used instead?

dimhotepus commented:
> Hello, thanks for your PR, I'll review it in chunks during this days, just to know, you runned the latest version of PVS Studio right? (I never used that program)

Well, at least after https://github.com/GameProgressive/UniSpySDK/pull/27/commits/2e2810142253f6f4455948b5d46c4ca852294312 SDK can be build.

arves100 commented:
> 
> 
> > Hello, thanks for your PR, I'll review it in chunks during this days, just to know, you runned the latest version of PVS Studio right? (I never used that program)
> 
> Yep, the latest one. Unfortunately i can't check the solution builds, as some functions are missed (Sha1*, MD5*, etc.) May be GS* equivalents should be used instead?

I suppose I forgot to push some changes I was doing a sort of big refactor of the whole SDK (so I can safetly wipe the extra platforms that shouldn't be there like nitro) so perhaps during an update to the OpenSDK I forgot to push the changes to the equivalents (which after this PR it's going to be cool having to retouch everything lmao)

arves100 commented on Chat/chatHandlers.c:
```diff
@@ -99,7 +99,7 @@
 
 #define FINISH_FILTER          ciFinishFilter(chat, filter, &params)
-#define IS_ALPHA(c) ((c >= 'A' && c <= 'Z') || (c >= 'a' && c <= 'z'))
+#define IS_ALPHA(c) (((c) >= 'A' && (c) <= 'Z') || ((c) >= 'a' && (c) <= 'z'))
```
ok

arves100 commented on common/darray.c:
```diff
@@ -114,7 +114,6 @@ void ArrayFree(DArray array)
 	}
 	// mj to do: move these asserts into gsi_free.  maybe, depends on whether user overloads them
 	GS_ASSERT(array->list);
-	GS_ASSERT(array);
```
ok

arves100 commented on Chat/chatMain.c:
```diff
@@ -2833,7 +2834,13 @@ void chatSetGlobalKeysA(CHAT chat,
+		// Do not overflow buffer with large key value pairs.  -1 for '\0'.
+		if (sizeof(buffer) - bufferLen - 1 < 2 + strlen(key) + strlen(value)) {
+			GS_ASSERT(0);
+			break;
```
I would put a return here

dimhotepus replied:
https://github.com/GameProgressive/UniSpySDK/pull/27/commits/b08895af7c572816b61cbea7a7fd728b61af0304

arves100 commented on Chat/chatMain.c:
```diff
@@ -3019,6 +3027,12 @@ void chatSetChannelKeysA(CHAT chat,
+		// Do not overflow buffer with large key value pairs.  -1 for '\0'.
+		if (sizeof(buffer) - bufferLen - 1 < 2 + strlen(keys[i]) + strlen(value)) {
+			GS_ASSERT(0);
+			break;
```
I would put a return here

dimhotepus replied:
https://github.com/GameProgressive/UniSpySDK/pull/27/commits/b08895af7c572816b61cbea7a7fd728b61af0304

arves100 commented on common/gsXML.c:
```diff
@@ -160,13 +160,13 @@ GSXmlStreamWriter gsXmlCreateStreamWriter(const char ** namespaces, int count)
 	{
 		GS_ASSERT(GS_XML_SOAP_DEFAULT_NAMESPACES[i] != NULL);
-		namespaceLen += strlen(GS_XML_SOAP_NAMESPACE_PREFIX)+1; // +1 for space
+		namespaceLen += sizeof(GS_XML_SOAP_NAMESPACE_PREFIX)/sizeof(GS_XML_SOAP_NAMESPACE_PREFIX[0])-1+1; // -1 for '\0', +1 for space
```
remove the comment and -1+1

dimhotepus replied:
https://github.com/GameProgressive/UniSpySDK/pull/27/commits/41205d7cea2bbfd8fddc847d7987bb0a1358c60d

arves100 commented on common/gsXML.c:
```diff
@@ -160,13 +160,13 @@ GSXmlStreamWriter gsXmlCreateStreamWriter(const char ** namespaces, int count)
 	{
 		GS_ASSERT(namespaces[i] != NULL);
-		namespaceLen += strlen(GS_XML_SOAP_NAMESPACE_PREFIX)+1; // +1 for space
+		namespaceLen += sizeof(GS_XML_SOAP_NAMESPACE_PREFIX)/sizeof(GS_XML_SOAP_NAMESPACE_PREFIX[0])-1+1; // -1 for '\0', +1 for space
```
remove the comment and -1+1

dimhotepus replied:
https://github.com/GameProgressive/UniSpySDK/pull/27/commits/41205d7cea2bbfd8fddc847d7987bb0a1358c60d

arves100 commented on GP/gp.c:
```diff
@@ -1322,7 +1322,7 @@ GPResult gpGetInfoNoWait(
 	if(iconnection->simulation)
 	{
-		memset(arg, 0, sizeof(arg));
+		memset(arg, 0, sizeof(*arg));
```
ok

arves100 commented on common/gsLargeInt.c:
```diff
@@ -1498,7 +1507,7 @@ gsi_bool gsiLargeIntMultM(gsLargeInt_t *x, gsLargeInt_t *y, const gsLargeInt_t *
 		if (gsi_is_false(gsiLargeIntSub(m->mData, m->mLength, &temp[logB_r], tempLen - logB_r, dest->mData, &dest->mLength)))
 		{
-			memset(temp, 0, sizeof(temp));
+			SafeZeroMemory(temp, sizeof(temp));
```
I would change the function name to gsiZeroMemory and move it's delcaration elsewhere (not in gsLargeInt) so other files could use it. common\gsMemory.c/h would be a good place

dimhotepus replied:
https://github.com/GameProgressive/UniSpySDK/pull/27/commits/9c1ad7bf869cd1e6da8fb92c464b8176264ccfc4

arves100 commented on GP/gpiProfile.c:
```diff
@@ -1071,12 +1071,12 @@ GPResult gpiAddToBlockedList(GPConnection * connection,
                 ArrayFree(profile->buddyStatusInfo->extendedInfoKeys);
                 profile->buddyStatusInfo->extendedInfoKeys = NULL;
             }
+            freeclear(profile->buddyStatusInfo);
```
ok

arves100 commented on GP/gpiBuddy.c:
```diff
@@ -1039,12 +1039,12 @@ gpiDeleteBuddy(
 			ArrayFree(pProfile->buddyStatusInfo->extendedInfoKeys);
 			pProfile->buddyStatusInfo->extendedInfoKeys = NULL;
 		}
+		freeclear(pProfile->buddyStatusInfo);
```
ok

arves100 commented on common/gsMemory.c:
```diff
@@ -660,11 +660,11 @@ void MEM_CHUNK_POOLSplitChunk(MEM_CHUNK_POOL *_this, MEM_CHUNK *header, gsi_bool
 	#if (MEM_PROFILE)
 		if(ReAlloc)
 		{
+			GS_ASSERT(_this->MemUsed >= MEM_CHUNKChunkSizeGet(header));
```
ok

arves100 commented on common/gsMemory.c:
```diff
@@ -912,8 +912,8 @@ void MEM_CHUNK_POOLFreeChunk(MEM_CHUNK_POOL *_this,MEM_CHUNK *header)
 	MEM_CHUNK *PrevFree;
 
 	#if (MEM_PROFILE)
+		GS_ASSERT(_this->MemUsed >= MEM_CHUNKMemUsedGet(header));
```
ok

arves100 commented on common/gsPlatformUtil.c:
```diff
@@ -1559,7 +1559,7 @@ char * gsiXxteaAlg(const char *sIn, int nIn, char key[XXTEA_KEY_SIZE], int bEnc,
 	
 	
-	oStr = (char *)gsimalloc((size_t)(4 * nIn + 1));
+	oStr = (char *)gsimalloc((4 * (size_t)nIn + 1));
```
same as above

dimhotepus replied:
duplicate of https://github.com/GameProgressive/UniSpySDK/pull/27#discussion_r674395408, will be resolved together.

arves100 commented on GP/gp.c:
```diff
@@ -1648,7 +1648,8 @@ GPResult gpGetBuddyStatus(
 	if(iconnection->simulation)
 	{
-		memset(status, 0, sizeof(GPBuddyStatus));
+		if(status != NULL)
```
While there might a null pointer error without this guard, the next line specifies that status must not be null.

I would remove this check and just move 
	// Check for a NULL status.
	///////////////////////////
	if(status == NULL)
		Error(connection, GP_PARAMETER_ERROR, "Invalid status.");
before this simulation block

dimhotepus replied:
Not sure with this, as looks like in simulation mode we never set error and always return `GP_NO_ERROR`.

`Error` in such case replaced with `{gpiSetErrorString(connection, string); return (result);}`

arves100 replied:
Hmm you're right here, the simulation mode should not return any error (well it should not even used to begin with)

arves100 commented on GP/gpiOperation.c:
```diff
@@ -21,9 +21,11 @@ gpiFailedOpCallback(
 {
 	GPICallback callback;
+	
+	GS_ASSERT(connection != NULL);
```
ok

arves100 commented on GP/gpiPeer.c:
```diff
@@ -1016,13 +1016,15 @@ gpiPeerStartTransferMessage(
 		return GP_NETWORK_ERROR;
+
+	tid.count = transferID->count;
+	tid.profileid = transferID->profileid;
```
ok

arves100 commented on GP/gpiPeer.c:
```diff
@@ -506,10 +506,10 @@ gpiProcessPeerConnected(
 	GS_ASSERT(peer);
 	if (!peer)
 		return;
+	GPIPeerOp *anIterator = peer->peerOpQueue.first;
```
ok

arves100 commented on GP/gpiBuffer.c:
```diff
@@ -206,7 +206,7 @@ gpiSendData(
 		{
 			static int sendCount;
-			char *buf = (char *)gsimalloc((size_t)(rcode + 1));
+			char *buf = (char *)gsimalloc((size_t)rcode + 1);
```
ok

arves100 commented on GP/gpiBuffer.c:
```diff
@@ -403,7 +403,7 @@ gpiRecvToBuffer(
 			{
 				static int recvCount;
-				char *buf = (char *)gsimalloc((size_t)(rcode + 1));
+				char *buf = (char *)gsimalloc((size_t)rcode + 1);
```
ok

arves100 commented on common/gsPlatformUtil.c:
```diff
@@ -1559,7 +1559,7 @@ char * gsiXxteaAlg(const char *sIn, int nIn, char key[XXTEA_KEY_SIZE], int bEnc,
 	
 	
-	oStr = (char *)gsimalloc((4 * (size_t)nIn + 1));
+	oStr = (char *)gsimalloc(4 * (size_t)nIn + 1);
```
duplicated? same as https://github.com/GameProgressive/UniSpySDK/pull/27/commits/aacbb3061e11398dc5293f5d8ba38debd1f576f8

dimhotepus replied:
yep, thanks.

arves100 commented on GP/gpiConnect.c:
```diff
@@ -336,8 +336,8 @@ gpiSendLogin(GPConnection * connection,
 	GPIProfile * profile;
 	char * passphrase;
-	char userBuffer[GP_NICK_LEN + GP_EMAIL_LEN];
 	char partnerBuffer[11];
```
I would do a macro GP_PARTNETBUFFER_LEN 11 and change this line to GP_NICK_LEN + GP_EMAIL_LEN + GP_PARTNERBUFFER_LENGTH + 1

dimhotepus replied:
https://github.com/GameProgressive/UniSpySDK/pull/27/commits/45fbbc5847c681a8cacaa983dafacdcc9627c4c8

arves100 commented on GP/gpiProfile.c:
```diff
@@ -111,8 +111,6 @@ static GPResult gpiOpenDiskProfiles(GPConnection * connection,
 
 	return GP_NO_ERROR;
-	
-	GSI_UNUSED(write);
```
ok

arves100 commented on common/gsUdpEngine.c:
```diff
@@ -1008,8 +1008,8 @@ GSUdpErrorCode gsUdpEngineAddMsgHandler(char theInitMsg[GS_UDP_MSG_HEADER_LEN],
 	GS_ASSERT(aUdp->mInitialized);
-	GS_ASSERT(theInitMsg || theInitMsg[0]);
-	GS_ASSERT(theHeader || theHeader[0]);
+	GS_ASSERT(theInitMsg && theInitMsg[0]);
```
ok

arves100 commented on ghttp/ghttpProcess.c:
```diff
@@ -1247,7 +1247,7 @@ static int ghiParseChunkSize
 	char * header;
 	int len;
-	int num;
+	unsigned num;
```
use "unsigned int" not unsigned

dimhotepus replied:
https://github.com/GameProgressive/UniSpySDK/pull/27/commits/bd64fbac79ff64d0bf18e90899d269f6d4401f50

arves100 commented on ghttp/ghttpProcess.c:
```diff
@@ -1257,10 +1257,10 @@ static int ghiParseChunkSize
 		return -1;
 
-	return num;
+	return (int)num;
```
According to https://en.wikipedia.org/wiki/Chunked_transfer_encoding, the chunk data must be >-1, I would change the function ghiParseChunkSize to be unsigned int and remove this cast

dimhotepus replied:
Problem here is `-1` value used as error indicator.

If we return `UINT_MAX` as error, it requires to change caller's `connection->chunkBytesLeft` type, which is `int`.
Which in turn requires to change internal parameters to `unsigned int` and `len` in `ghiProcessIncomingFileData`, and so on.

arves100 replied:
hmm true I forgot to see the use of that (whicle perhaps 0 might be an error as well 🤔 ), better to keep it this way then

arves100 commented on common/gsSoap.c:
```diff
@@ -212,7 +212,7 @@ gsi_bool gsiSoapGetSessionTokenFromHeaders(const char *headers, char valueBuffer
 
 		// scan in the result
-		rcode = sscanf(header, " %s", valueBuffer);
+		rcode = sscanf(header, " %36s", valueBuffer);
```
ok

arves100 commented on ghttp/ghttp.h:
```diff
@@ -197,7 +197,7 @@ typedef int GHTTPRequest;
 #endif
 
-#define IS_GHTTP_ERROR(x) (x<0)
+#define IS_GHTTP_ERROR(x) ((x)<0)
```
ok

arves100 commented on ghttp/ghttpProcess.c:
```diff
@@ -1779,7 +1779,7 @@ void ghiDoReceivingHeaders
                 {
                     // Same length, maybe a bigger number
-                    if( strncmp(pStart,szMaxSize,(unsigned int)(pEnd-pStart)) >= 0 )
+                    if( strncmp(pStart,szMaxSize,(size_t)(pEnd-pStart)) >= 0 )
```
ok

arves100 commented on Peer/peerMain.h:
```diff
@@ -52,8 +52,8 @@ piConnection * connection;  // This is to fool Visual Assist.
 
-#define ASSERT_ROOMTYPE(type)     GS_ASSERT((type == TitleRoom) || (type == GroupRoom) || (type == StagingRoom))
-#define ASSERT_MESSAGETYPE(type)  GS_ASSERT((type == NormalMessage) || (type == ActionMessage) || (type == NoticeMessage))
+#define ASSERT_ROOMTYPE(type)     GS_ASSERT(((type) == TitleRoom) || ((type) == GroupRoom) || ((type) == StagingRoom))
```
ok

arves100 commented on qr2/qr2.c:
```diff
@@ -1648,7 +1648,7 @@ static void send_keepalive(qr2_t qrec)
 
 	gsDebugFormat(GSIDebugCat_QR2, GSIDebugType_Network, GSIDebugLevel_Comment,
-		"Sent keepalive to master\r\n", buf);
+		"Sent keepalive to master (size %d)\r\n", buf.len);
```
ok

arves100 commented on sc/sciReport.c:
```diff
@@ -354,9 +354,12 @@ SCResult SC_CALL sciReportBeginNewTeam(SCIReport * theReport)
+
 	GS_ASSERT(theReport != NULL);
 
+	aHeader = (SCIReportHeader*)theReport->mBuffer.mData;
```
ok

arves100 commented on sc/sciReport.c:
```diff
@@ -391,9 +394,12 @@ SCResult SC_CALL sciReportBeginNewPlayer(SCIReport * theReport)
 
 	GS_ASSERT(theReport != NULL);
+	
+	theReport = (SCIReportHeader*)theReport->mBuffer.mData;
```
ok

arves100 commented on gstats/gstats.c:
```diff
@@ -215,7 +215,7 @@ char *GenerateAuthW(const char* challenge, const unsigned short *password, char
 {
 	struct sockaddr_in saddr;
-	char tempHostname[128];
+	char tempHostname[sizeof(gcd_gamename)/sizeof(gcd_gamename[0])];
```
256+64+2 (gcd_gamename is not enough)

dimhotepus replied:
https://github.com/GameProgressive/UniSpySDK/pull/27/commits/7fd1270ff84075697b4514d57da20aeffc5174dd

arves100 commented on gstats/gbucket.c:
```diff
@@ -210,7 +210,7 @@ void *BucketConcat(bucketset_t set, char *name, void *value)
 }
 
-#define AVG(cur, new, num) (((cur * num) + new) / (++num))
+#define AVG(cur, new, num) ((((cur) * (num)) + (new)) / (++num))
```
ok

arves100 commented on gstats/gstats.c:
```diff
@@ -958,7 +958,7 @@ static int SendChallengeResponse(const char *indata, int gameport)
 	static char challengestr[] = {'\0','h','a','l','l','e','n','g','e','\0'};
 	char *challenge;
-	char resp[128];
+	char resp[256];
```
384 (better if this would be modified in the future to compute the correct key length)

dimhotepus replied:
https://github.com/GameProgressive/UniSpySDK/pull/27/commits/dd1ea8b03016b32abcb8f71cf8f4ef6792440cd8

arves100 commented on webservices/AuthService.c:
```diff
@@ -1143,7 +1143,7 @@ gsi_bool wsLoginCertReadBinary(GSLoginCertificate * certOut, char * bufin, unsig
 	READ_NTS(certOut->mProfileNick, WS_LOGIN_NICK_LEN);
 	READ_NTS(certOut->mUniqueNick, WS_LOGIN_UNIQUENICK_LEN);
-	READ_NTS(certOut->mCdKeyHash, WS_LOGIN_CDKEY_LEN);
+	READ_NTS(certOut->mCdKeyHash, WS_LOGIN_KEYHASH_LEN);
```
ok

arves100 commented on webservices/AuthService.c:
```diff
@@ -613,7 +613,7 @@ static void wsLoginRemoteAuthCallback(GHTTPResult theResult,
 				}
 
-				sprintf(buffer, "%d", cert->mProfileId);
+				sprintf(buffer, "%u", cert->mProfileId);
```
ok

arves100 commented on webservices/AuthService.c:
```diff
@@ -427,7 +427,7 @@ static void wsLoginUniqueCallback(GHTTPResult theResult,
 				}
 
-				sprintf(buffer, "%d", cert->mProfileId);
+				sprintf(buffer, "%u", cert->mProfileId);
```
ok

arves100 commented on webservices/AuthService.c:
```diff
@@ -240,7 +240,7 @@ static void wsiLoginProfileCallback(GHTTPResult theResult,
 				}
 
-				sprintf(buffer, "%d", cert->mProfileId);
+				sprintf(buffer, "%u", cert->mProfileId);
```
ok

arves100 commented on webservices/AuthService.c:
```diff
@@ -71,7 +71,7 @@ const char WS_AUTHSERVICE_SIGNATURE_EXP[] =
 
 char wsAuthServiceURL[WS_LOGIN_MAX_URL_LEN] = "";
-char authCreds[54];
+char authCreds[92];
```
ok

arves100 commented on GP/gpi.c:
```diff
@@ -20,11 +20,11 @@
-void MD5Final(unsigned char [16], MD5_CTX *);
-void MD5Print(unsigned char [16], char[33]);
-void MD5Digest(unsigned char *, unsigned int, char[33]);
+void GSMD5Init(GSMD5_CTX *);
```
k

arves100 commented on GP/gpiConnect.c:
```diff
@@ -710,7 +710,7 @@ gpiProcessConnect(
 			data->userChallenge,
 			data->passwordHash);
-		MD5Digest((unsigned char *)buffer, strlen(buffer), check);
+		GSMD5Digest((unsigned char *)buffer, strlen(buffer), check);
```
ok

arves100 commented on common/gsCrypt.c:
```diff
@@ -78,7 +78,7 @@
 				seedPlusIter[GS_CRYPT_RSA_DATABLOCKSIZE+2] = 0x00;
 				seedPlusIter[GS_CRYPT_RSA_DATABLOCKSIZE+3] = (gsi_u8)(i/GS_CRYPT_HASHSIZE);
-				MD5Digest(seedPlusIter, seedLen+sizeof(gsi_u32), hashHexStr);
+				GSMD5Digest(seedPlusIter, seedLen+sizeof(gsi_u32), hashHexStr);
```
ok

arves100 commented on common/md5.h:
```diff
@@ -67,13 +67,13 @@ typedef struct {
   UINT4 count[2];                /* number of bits, modulo 2^64 (lsb first) */
   unsigned char buffer[64];                                 /* input buffer */
-} MD5_CTX;
+} GSMD5_CTX;
```
ok

arves100 commented on common/md5c.c:
```diff
@@ -101,7 +101,7 @@ static unsigned char PADDING[64] = {
 /* MD5 initialization. Begins an MD5 operation, writing a new context.
  */
-void MD5Init (MD5_CTX *context)
+void GSMD5Init (GSMD5_CTX *context)
```
ok

arves100 commented on GP/gp.c:
```diff
@@ -171,7 +171,7 @@ GPResult gpConnectA(
 	// Do it.
 	/////////
-	return gpiConnect(connection, nick, "", email, password, "", "", "", NULL, firewall, GPIFalse, blocking, callback, param);
+	return gpiConnect(connection, nick, "", email, password, "", "", "", firewall, GPIFalse, blocking, callback, param);
```
ok

arves100 commented on gcdkey/gcdkeyc.c:
```diff
@@ -51,12 +53,17 @@ void gcd_compute_response(char *cdkey, char *challenge, char response[RESPONSE_S
+
 	/* do the cd key md5 */
-	GSMD5Digest((unsigned char *)cdkey, strlen(cdkey), response);
+	GSMD5Digest((unsigned char *)cdkey, (unsigned)cdKeyLen, response);
```
use unsigned int

dimhotepus replied:
https://github.com/GameProgressive/UniSpySDK/pull/27/commits/bd64fbac79ff64d0bf18e90899d269f6d4401f50

arves100 commented on gcdkey/gcdkeyc.c:
```diff
@@ -51,12 +53,17 @@ void gcd_compute_response(char *cdkey, char *challenge, char response[RESPONSE_S
 	gsiSafeStrcpyA(&response[32], randstr, RESPONSE_SIZE - 32);
 	/* do the response md5 */
-	GSMD5Digest((unsigned char *)rawout, strlen(rawout), &response[40]);	
+	GSMD5Digest((unsigned char *)rawout, (unsigned)rawoutLen, &response[40]);
```
use usnigned int

dimhotepus replied:
https://github.com/GameProgressive/UniSpySDK/pull/27/commits/bd64fbac79ff64d0bf18e90899d269f6d4401f50

arves100 commented on gcdkey/gcdkeyc.c:
```diff
@@ -51,12 +53,17 @@ void gcd_compute_response(char *cdkey, char *challenge, char response[RESPONSE_S
+	rawoutLen = strlen(rawout);
+
+	GS_ASSERT(cdKeyLen <= UINT_MAX);
+	GS_ASSERT(rawoutLen <= UINT_MAX);
```
ok

arves100 commented on gcdkey/gcdkeys.c:
```diff
@@ -822,15 +822,16 @@ static void resend_auth_req(gsclient_t *client)
-		xcode_buf(buf, strlen(keepAlive));
-		sendto(sock, buf, strlen(keepAlive), 0, (struct sockaddr *)&valaddr, sizeof(struct sockaddr_in));
+		xcode_buf(buf, keepAliveLen);
+		sendto(sock, buf, keepAliveLen, 0, (struct sockaddr *)&valaddr, sizeof(struct sockaddr_in));
```
ok

arves100 commented on Chat/chatHandlers.c:
```diff
@@ -21,15 +21,6 @@
 #include "chatCallbacks.h"
 #include "chatCrypt.h"
 
-#if defined(_WIN32)
```
ok

arves100 commented on Chat/chatChannel.c:
```diff
@@ -17,11 +17,6 @@
 
-#if defined(_WIN32)
-// Compiler warns when explicitly casting from function* to void*
-#pragma warning(disable:4054)
```
ok

arves100 commented on Chat/chatCallbacks.c:
```diff
@@ -14,11 +14,6 @@
 #include "chatChannel.h"
 
-#if defined(_WIN32)
-// warning about casting void* to function*
```
ok

arves100 commented on Chat/chatSocket.c:
```diff
@@ -982,9 +985,13 @@ static CHATBool ciParseInput(ciSocket * sock)
 				// Take the message out of the buffer.
 				//////////////////////////////////////
-				ciBufferClipFront(&sock->inputQueue, (q - sock->inputQueue.buffer));
+				ciBufferClipFront(&sock->inputQueue, (int)messageLen);
```
ok

arves100 commented on GP/gpiBuddy.c:
```diff
@@ -1051,7 +1054,7 @@ gpiDeleteBuddy(
 		assert(iconnection->profileList.numBuddies >= 0);
 #ifndef _PS2
-		gpiProfileMap(connection, gpiFixBuddyIndices, (void *)(unsigned long)index);
+		gpiProfileMap(connection, gpiFixBuddyIndices, (void *)(intptr_t)index);
```
ok

arves100 commented on GP/gpiConnect.c:
```diff
@@ -350,7 +351,11 @@ gpiSendLogin(GPConnection * connection,
+	passphraseLen = strlen(passphrase);
+	GS_ASSERT(passphraseLen <= UINT_MAX);
+
+	GSMD5Digest((unsigned char*)passphrase, (unsigned int)passphraseLen, data->passwordHash);
```
ok

arves100 commented on GP/gpiProfile.c:
```diff
@@ -1151,7 +1155,7 @@ GPResult gpiRemoveFromBlockedList(GPConnection * connection,
 
 #ifndef _PS2
-        gpiProfileMap(connection, gpiFixBlockIndices, (void *)(unsigned long)index);
+        gpiProfileMap(connection, gpiFixBlockIndices, (void *)(intptr_t)index);
```
ok

arves100 commented on GP/gpiTransfer.c:
```diff
@@ -1348,7 +1348,9 @@ static GPResult gpiSendFileData
-	gpiFinishTransferMessage(connection, transfer, (char *)data, len);
+	GS_ASSERT(len <= INT_MAX);
+
+	gpiFinishTransferMessage(connection, transfer, (char *)data, (int)len);
```
ok

arves100 commented on GP/gpiConnect.c:
```diff
@@ -386,7 +391,7 @@ gpiSendLogin(GPConnection * connection,
 		data->serverChallenge,
 		data->passwordHash);
-	GSMD5Digest((unsigned char *)buffer, strlen(buffer), response);
+	GSMD5Digest((unsigned char *)buffer, (unsigned int)strlen(buffer), response);
```
ok

arves100 commented on GP/gpiConnect.c:
```diff
@@ -710,7 +715,7 @@ gpiProcessConnect(
 			data->userChallenge,
 			data->passwordHash);
-		GSMD5Digest((unsigned char *)buffer, strlen(buffer), check);
+		GSMD5Digest((unsigned char *)buffer, (unsigned int)strlen(buffer), check);
```
ok

arves100 commented on GP/gpiPeer.c:
```diff
@@ -253,7 +253,7 @@ gpiProcessPeerAcceptingConnection(
 				iconnection->profileid,
 				pid);
-			GSMD5Digest((unsigned char *)buffer, strlen(buffer), sigCheck);
+			GSMD5Digest((unsigned char *)buffer, (unsigned int)strlen(buffer), sigCheck);
```
ok

arves100 commented on GP/gpiBuffer.c:
```diff
@@ -658,7 +658,7 @@ gpiReadMessageFromBuffer(
 			// Set the position to the end of the message.
 			//////////////////////////////////////////////
-			inputBuffer->pos = ((str - inputBuffer->buffer) + len + 1);
+			inputBuffer->pos = (int)((str - inputBuffer->buffer) + len + 1);
```
ok for now, in the future I would change positions+length of buffers to uint

arves100 commented on GP/gpi.c:
```diff
@@ -350,7 +350,7 @@ gpiProcessConnectionManager(
 			// Move the rest of the connect buffer up to the front.
 			///////////////////////////////////////////////////////
-			iconnection->socketBuffer.len -= (next - iconnection->socketBuffer.buffer);
+			iconnection->socketBuffer.len -= (int)(next - iconnection->socketBuffer.buffer);
```
ok for now, in the future I would change positions+length of buffers to uint

arves100 commented on GP/gpiUnique.c:
```diff
@@ -123,6 +123,9 @@ gpiSendRegisterCdKey(
 )
 {
+	// TODO: Looks like some code using gameId not ported from Open SDK.
+	GSI_UNUSED(gameId);
```
ok for now untill the new gameId param is ported

arves100 reviewed (changes requested):
I've reviewed up to https://github.com/GameProgressive/UniSpySDK/pull/27/commits/3078da53f97fda04deb6a8ceec4978e15e81e65f.

Those commits requires some adjustment:
https://github.com/GameProgressive/UniSpySDK/pull/27/commits/aacbb3061e11398dc5293f5d8ba38debd1f576f8
https://github.com/GameProgressive/UniSpySDK/pull/27/commits/da1d314e0e18fbd7c31c16747dede11dbc443946
https://github.com/GameProgressive/UniSpySDK/pull/27/commits/47230b3c5d9993713f34748f1f0a1e8348f8a056
https://github.com/GameProgressive/UniSpySDK/pull/27/commits/2ff428ea6cd38f31b48c3a4d6f12f38718b7739d
https://github.com/GameProgressive/UniSpySDK/pull/27/commits/d3fcf36a16de9134011bd234eb7d0a6a69b1638d
https://github.com/GameProgressive/UniSpySDK/pull/27/commits/3bef25df661ac64e108d213acdfcdd292d38b08a
https://github.com/GameProgressive/UniSpySDK/pull/27/commits/faff80e6f3dd845325e77079a505bd8c550b7a82
https://github.com/GameProgressive/UniSpySDK/pull/27/commits/24ea6d42ab5ca455a5f0a1e6aa144973e13e52cd
https://github.com/GameProgressive/UniSpySDK/pull/27/commits/5512fac5d92b1043d62a07f483f60f48fde0d4da
https://github.com/GameProgressive/UniSpySDK/pull/27/commits/643ec3000cdcc1fca21daa303127b8064f82c8c7
https://github.com/GameProgressive/UniSpySDK/pull/27/commits/fd27de0892c79a9c22dfe7e3392e1f6610365b8f
https://github.com/GameProgressive/UniSpySDK/pull/27/commits/e85e7453f068572801e26d652f20abe18755ad34

dimhotepus commented on common/gsPlatformUtil.c:
```diff
@@ -1507,7 +1507,7 @@ char * gsiXxteaAlg(const char *sIn, int nIn, char key[XXTEA_KEY_SIZE], int bEnc,
 
 	// Load and zero-pad entire input stream as 32-bit words
-	sIn2 = (char *)gsimalloc((size_t)(4 * nIn));
+	sIn2 = (char *)gsimalloc((4 * (size_t)nIn));
```
Not feel safe with such change, as it: 
* creates additional mix of signed / unsigned arithmetic.
* requires conversion `int` => `ptrdiff_t` for internal parameters.
* algorithm may rely on internal `unsigned int` wrap.

Suggest to not convert to `size_t` for now. Added fix for memory leak (never occurs, but still) in https://github.com/GameProgressive/UniSpySDK/pull/27/commits/bd7db12d2256d18cf024c36d8da00e268e0c96ab.

arves100 replied:
hmm makes sense to keep it this way as you pointed out, lgtm

dimhotepus commented on Chat/chatCallbacks.c:
```diff
@@ -562,10 +562,12 @@ CHATBool ciAddCallback_(CHAT chat, int type, void * callback, void * callbackPar
 	if(data.callbackParams == NULL)
 		return CHATFalse; //ERRCON
+#ifdef _DEBUG
+	memset(data.callbackParams, 0xC4, callbackParamsSize + 64);
```
https://github.com/GameProgressive/UniSpySDK/pull/27/commits/ce6ac2d09822a764f6cf51d522c0a216cbc7f1b6

dimhotepus commented:
@arves100 Thank you for review. Added fixes and some notes. 

Also added some interesting findings by MSVC and think it is enough for this PR (already too large) :).

arves100 commented on ghttp/ghttpProcess.c:
```diff
@@ -1208,7 +1208,7 @@ static GHTTPBool ghiDeliverIncomingFileData
 		int bytesWritten = 0;
 #ifndef NOFILE
-		bytesWritten = fwrite(data, 1, len, connection->saveFile);
+		bytesWritten = (int)fwrite(data, 1, len, connection->saveFile);
```
doesn't fwrite return size_t?

dimhotepus replied:
`len` is `int`, `count` is `1`, so `1 * len` is always `int`. 
`bytesWritten` <= `1 * len` so decided to be less intrusive and use `int`.

May change to `size_t` for `len` argument and so on, but that requires more changes.

arves100 replied:
hmm by giving a second look, the people at gamespy decided to use all the lengths as "int", so it doesn't currently make a lot of sense to use size_t atm, it's ok to keep it this way

arves100 commented on webservices/AuthService.c:
```diff
@@ -895,9 +894,17 @@ static void wsiLoginEncryptPassword(const gsi_char * password, gsi_u8 ciphertext
+
+	GS_ASSERT(passwordLen <= UINT_MAX);
+
+	gsCryptRSAEncryptBuffer(&sigkeypub, (const gsi_u8*)password, (unsigned int)passwordLen, ciphertext);
```
ok

arves100 commented on webservices/AuthService.c:
```diff
@@ -956,9 +963,9 @@ gsi_bool wsLoginCertIsValid(const GSLoginCertificate * cert)
-	GSMD5Update(&md5, (unsigned char*)&cert->mProfileNick, strlen(cert->mProfileNick)); 
-	GSMD5Update(&md5, (unsigned char*)&cert->mUniqueNick, strlen(cert->mUniqueNick));   
-	GSMD5Update(&md5, (unsigned char*)&cert->mCdKeyHash, strlen(cert->mCdKeyHash));     
+	GSMD5Update(&md5, (unsigned char*)&cert->mProfileNick, (unsigned int)strlen(cert->mProfileNick)); 
```
ok

arves100 commented on webservices/AuthService.c:
```diff
@@ -992,7 +999,7 @@ gsi_bool wsLoginCertIsValid(const GSLoginCertificate * cert)
                                       return gsi_false; \
 								  strcpy(bufout+lenoutSoFar, a); \
-								  lenoutSoFar += _tcslen(a) + 1; }
+								  lenoutSoFar += (unsigned int)_tcslen(a) + 1; }
```
please use gsi_u32 for all the casts in lenoutSoFar

dimhotepus replied:
https://github.com/GameProgressive/UniSpySDK/pull/27/commits/fafe1ceeb163b4ba77da8c57e3dec9a3f2ad6e2d

arves100 commented on Voice2/gvDirectSound.c:
```diff
@@ -5,7 +5,6 @@
 #include "gvCodec.h"
 #include "gvSource.h"
 #include "gvUtil.h"
-#pragma warning(disable:4201)
```
ok

arves100 commented on gstats/gbucket.c:
```diff
@@ -344,16 +347,15 @@ static bucket_t *DoFind(bucketset_t set, char *name)
 
 	char *s = ((bucket_t *)elem)->name;
-    len = strlen(s);
+	len = strlen(s);
```
ok

arves100 commented on pt/ptMain.c:
```diff
@@ -602,7 +602,7 @@ static GHTTPBool ptaFilePlanetCompletedCallback
 		// Copy off the name.
 		/////////////////////
-		len = (str - Line);
+		len = (int)(str - Line);
```
ok

arves100 commented on Peer/peerOperations.c:
```diff
@@ -580,7 +581,11 @@ PEERBool piNewConnectOperation
+
+	GS_ASSERT(uniqueIDLen <= UINT_MAX);
+
+	GSMD5Digest((unsigned char *)uniqueID, (unsigned int)uniqueIDLen, encodedUniqueID);
```
ok

arves100 commented on common/gsPlatformSocket.c:
```diff
@@ -725,9 +716,3 @@ gsi_u32 gsiGetBroadcastIP(void)
 ///////////////////////////////////////////////////////////////////////////////
 ///////////////////////////////////////////////////////////////////////////////
-
-// Re-enable previously disabled compiler warnings
```
ok

arves100 commented on common/gsPlatformUtil.c:
```diff
@@ -1886,7 +1886,7 @@ const char * GOAGetUniqueID_Internal(void)
 			ret = RegCreateKeyExA(HKEY_CURRENT_USER, REG_KEY, 0, NULL, REG_OPTION_NON_VOLATILE, KEY_ALL_ACCESS, NULL, &thekey, &disp);
 		}
-		RegSetValueExA(thekey, (LPCSTR)"Crypt", 0, REG_SZ, (const LPBYTE)keyval, strlen(keyval)+1);
+		RegSetValueExA(thekey, (LPCSTR)"Crypt", 0, REG_SZ, (const LPBYTE)keyval, (DWORD)strlen(keyval)+1);
```
ok

arves100 commented on common/gsPlatformUtil.c:
```diff
@@ -1555,10 +1555,16 @@ char * gsiXxteaAlg(const char *sIn, int nIn, char key[XXTEA_KEY_SIZE], int bEnc,
 	}
-	else return NULL;
+	else
+	{
```
ok

arves100 commented on Chat/chatCallbacks.c:
```diff
@@ -560,10 +560,10 @@ CHATBool ciAddCallback_(CHAT chat, int type, void * callback, void * callbackPar
+	memcpy(data.callbackParams, callbackParams, callbackParamsSize);
 #ifdef _DEBUG
-	memset(data.callbackParams, 0xC4, callbackParamsSize + 64);
+	memset((unsigned char*)data.callbackParams + callbackParamsSize, 0xC4, 64);
```
ok

arves100 commented on Chat/chatMain.c:
```diff
@@ -3028,7 +3028,7 @@ void chatSetChannelKeysA(CHAT chat,
 		if (sizeof(buffer) - bufferLen - 1 < 2 + strlen(keys[i]) + strlen(value)) {
 			GS_ASSERT(0);
-			break;
+			return;
```
ok

arves100 commented on common/gsXML.c:
```diff
@@ -160,13 +160,13 @@ GSXmlStreamWriter gsXmlCreateStreamWriter(const char ** namespaces, int count)
 	{
 		GS_ASSERT(namespaces[i] != NULL);
-		namespaceLen += sizeof(GS_XML_SOAP_NAMESPACE_PREFIX)/sizeof(GS_XML_SOAP_NAMESPACE_PREFIX[0])-1+1; // -1 for '\0', +1 for space
+		namespaceLen += sizeof(GS_XML_SOAP_NAMESPACE_PREFIX)/sizeof(GS_XML_SOAP_NAMESPACE_PREFIX[0]);
```
ok

arves100 commented on common/gsMemory.c:
```diff
@@ -119,6 +119,15 @@ static void* MEM_MANAGER_CALL _gsi_realloc(void* ptr, size_t size)
 
+// Prevent compiler from inlining memset and further removal if gsiZeroMemory buffer
+// is not accessed later.
+static volatile void* (*memset_func)(void*, int, size_t) = memset;
```
ok

arves100 commented on GP/gpiConnect.c:
```diff
@@ -540,8 +540,8 @@ gpiProcessConnect(
 	GPIProfile * profile;
-	char partnerBuffer[11];
-	char userBuffer[GP_NICK_LEN + GP_EMAIL_LEN + sizeof(partnerBuffer) / sizeof(partnerBuffer[0]) + 1];  // + 1 for @
+	char partnerBuffer[GP_PARNTERBUFFER_LEN];
```
ok

arves100 commented on ghttp/ghttpProcess.c:
```diff
@@ -1247,7 +1247,7 @@ static int ghiParseChunkSize
 	char * header;
 	int len;
-	unsigned num;
+	unsigned int num;
```
ok

arves100 commented on gstats/gstats.c:
```diff
@@ -215,7 +215,7 @@ char *GenerateAuthW(const char* challenge, const unsigned short *password, char
 {
 	struct sockaddr_in saddr;
-	char tempHostname[sizeof(gcd_gamename)/sizeof(gcd_gamename[0])];
+	char tempHostname[sizeof(gcd_gamename)/sizeof(gcd_gamename[0]) + 2 + sizeof(StatsServerHostname)/sizeof(StatsServerHostname[0])];
```
ok

arves100 commented on gstats/gstats.c:
```diff
@@ -958,7 +958,7 @@ static int SendChallengeResponse(const char *indata, int gameport)
 	static char challengestr[] = {'\0','h','a','l','l','e','n','g','e','\0'};
 	char *challenge;
-	char resp[256];
+	char resp[384];
```
ok

arves100 commented on gstats/gstats.c:
```diff
@@ -978,7 +978,7 @@ static int SendChallengeResponse(const char *indata, int gameport)
 	GSMD5Digest((unsigned char *)resp, (unsigned int)len, md5val);
 	DOXCODE(respformat, sizeof(respformat)-1, enc3);
-	len = sprintf(resp,respformat,gcd_gamename, md5val, gameport);
+	len = sprintf(resp,respformat,gcd_gamename, md5val, theGameport);
```
ok

arves100 commented on serverbrowsing/sb_serverlist.c:
```diff
@@ -1275,12 +1275,12 @@ static SBError ProcessMainListData(SBServerList *slist)
-			inbuf += keylen;
-			inlen -= keylen;
+			inbuf += valkeylen;
+			inlen -= valkeylen;
```
ok

arves100 commented on natneg/natneg.c:
```diff
@@ -592,16 +592,16 @@ static void NegotiateThink(NATNegotiator neg)
+			StartReport(neg, nr_success, neg->negotiateSock, (struct sockaddr_in *)&saddrReport);
 		else
-			StartReport(neg, nr_success, neg->gameSock, (struct sockaddr_in *)&saddr);
+			StartReport(neg, nr_success, neg->gameSock, (struct sockaddr_in *)&saddrReport);
```
ok

arves100 commented on Chat/chatHandlers.c:
```diff
@@ -2891,9 +2891,9 @@ void ciRplEndOfWhoHandler(CHAT chat, const ciServerMessage * message)
 }
 
-static char * ciParseValue(const char * flags, int * len)
+static char * ciParseValue(const char * flags, size_t * len)
```
this whole commit is ok

arves100 commented on GP/gpiPeer.c:
```diff
@@ -932,10 +932,10 @@ gpiPeerStartConnect(
 		GSUdpPeerState aPeerState;
 		gsUdpEngineGetPeerState(profile->buddyStatusInfo->buddyIp , profile->buddyStatusInfo->buddyPort, &aPeerState);
-		if (aPeerState != GS_UDP_PEER_CONNECTED || aPeerState != GS_UDP_PEER_CONNECTING)
+		if (aPeerState != GS_UDP_PEER_CONNECTED && aPeerState != GS_UDP_PEER_CONNECTING)
```
ok

arves100 commented on Peer/peerRooms.c:
```diff
@@ -295,7 +295,7 @@ void piSetLocalFlags
 )
 {
-	char buffer[NumRooms][128];
+	char buffer[NumRooms][128] = { {'\0'}, {'\0'}, {'\0'} };
```
ok

arves100 commented on Chat/chatSocket.c:
```diff
@@ -543,7 +543,7 @@ CHATBool ciSocketSendf(ciSocket * sock,
 	/////////////////////
 	va_start(args, format);
-	num = vsprintf(buffer, format, args);
+	num = vsnprintf(buffer, sizeof(buffer), format, args);
```
ok

arves100 commented on gstats/gstats.c:
```diff
@@ -1009,7 +1009,7 @@ static int RecvSessionKey()
 	}
 
-	resp[len] = 0;
+	resp[len - 1] = '\0';
```
ok

arves100 commented on common/gsDebug.c:
```diff
@@ -122,8 +122,10 @@ void gsDebugVaList(GSIDebugCategory theCat, GSIDebugType theType,
+	GS_ASSERT(theCat   >= GSIDebugCat_App);
+	GS_ASSERT(theCat   < GSIDebugCat_Count);
+	GS_ASSERT(theType  >= GSIDebugType_Network);
+	GS_ASSERT(theType  < GSIDebugType_Count);
```
ok

arves100 commented on common/gsPlatformUtil.c:
```diff
@@ -461,9 +461,15 @@ void SocketStartUp()
-		WSAStartup(MAKEWORD(2,2), &data);
+		if (0 != WSAStartup(MAKEWORD(2,2), &data))
+		{
+			OutputDebugString("WSAStartup(2, 2) failed\n");
```
Ok, this code will bne changed anyway in the future

arves100 commented on common/gsSHA1.c:
```diff
@@ -184,26 +184,28 @@ int GSSHA1Input(    GSSHA1Context    *context,
-    }
-
-    message_array++;
+      GS_ASSERT(context->Message_Block_Index < sizeof(context->Message_Block));
```
ok

arves100 commented on Peer/peerCallbacks.c:
```diff
@@ -3570,7 +3570,7 @@ static int piAddCallback
 
 	GS_ASSERT(callback);
-	//GS_ASSERT(type >= 0);
+	GS_ASSERT(type >= PI_CONNECT_CALLBACK);
```
ok

arves100 commented on GP/gpiConnect.c:
```diff
@@ -540,8 +540,8 @@ gpiProcessConnect(
 	GPIProfile * profile;
-	char userBuffer[GP_NICK_LEN + GP_EMAIL_LEN];
 	char partnerBuffer[11];
+	char userBuffer[GP_NICK_LEN + GP_EMAIL_LEN + sizeof(partnerBuffer) / sizeof(partnerBuffer[0]) + 1];  // + 1 for @
```
ok

arves100 commented on serverbrowsing/sb_serverbrowsing.c:
```diff
@@ -257,7 +257,7 @@ static SBError WaitForTriggerUpdate(ServerBrowser sb, SBBool viaMaster)
 		msleep(10);
 		err = ServerBrowserThink(sb);
-		if (viaMaster && sb->list.state == sb_disconnected) //we were supposed to get from master, and it's disconnected
+		if (viaMaster && sb->list.state == sl_disconnected) //we were supposed to get from master, and it's disconnected
```
hmm this looks ok, I would 100% verify it once it gets merged

arves100 commented on sc/sciReport.c:
```diff
@@ -397,10 +397,9 @@ SCResult SC_CALL sciReportBeginGlobalData(SCIReport * theReport)
-
 	GS_ASSERT(theReport->mReportState == SCIReportState_ROSTER);
+	
+	aHeader = (SCIReportHeader*)theReport->mBuffer.mData;
```
ok

arves100 reviewed (changes requested):
Thanks for all your commits! There should be an issue there (maybe even more sorry I kinda lost track of all the changes haha)
https://github.com/GameProgressive/UniSpySDK/pull/27/commits/3ff5871b6407d91b80fa85468cab847acef68fd4

Overall everything looks cool, once the last issues are fixed I will merge the changes.

dimhotepus commented on ghttp/ghttpPost.c:
```diff
@@ -1431,7 +1431,7 @@ static GHIPostingResult ghiPostStateDoPosting
 
 					// type
-					writen = gsiSafeStrcpyA(&buffer[writePos], contentType, sizeof(buffer) - writePos);
+					writen = (unsigned int)gsiSafeStrcpyA(&buffer[writePos], contentType, sizeof(buffer) - writePos);
```
Yep, looks like i mistaken here. Thank you!

https://github.com/GameProgressive/UniSpySDK/pull/27/commits/8e7d3280a89e855a070912b957ee81aec66d63de

dimhotepus commented:
Thank you for patience!

Remaining commits to discuss:
https://github.com/GameProgressive/UniSpySDK/pull/27/files/173f1fe722b448fb33fa78a210abb25d0dd3d410..aacbb3061e11398dc5293f5d8ba38debd1f576f8#diff-35ca133c051513dc17a5b8f9280e7feefb421d0156883802222a85130a72c382
https://github.com/GameProgressive/UniSpySDK/pull/27/files/3078da53f97fda04deb6a8ceec4978e15e81e65f..3ff5871b6407d91b80fa85468cab847acef68fd4#diff-c3ba31498f33cf4a31e709e3d08cd8afca849076dc387a2b2f51e834e2655785

and you was interested in

aacbb30
da1d314
47230b3
2ff428e
d3fcf36
3bef25d
faff80e
24ea6d4
5512fac
643ec30
fd27de0
e85e745

What if i split PR on three with roughly 25 commits per each? It will simplify review.

arves100 commented:
> 
> 
> Thank you for patience!
> 
> Remaining commits to discuss:
> https://github.com/GameProgressive/UniSpySDK/pull/27/files/173f1fe722b448fb33fa78a210abb25d0dd3d410..aacbb3061e11398dc5293f5d8ba38debd1f576f8#diff-35ca133c051513dc17a5b8f9280e7feefb421d0156883802222a85130a72c382
> https://github.com/GameProgressive/UniSpySDK/pull/27/files/3078da53f97fda04deb6a8ceec4978e15e81e65f..3ff5871b6407d91b80fa85468cab847acef68fd4#diff-c3ba31498f33cf4a31e709e3d08cd8afca849076dc387a2b2f51e834e2655785
> 
> and you was interested in
> 
> [aacbb30](https://github.com/GameProgressive/UniSpySDK/commit/aacbb3061e11398dc5293f5d8ba38debd1f576f8)
> [da1d314](https://github.com/GameProgressive/UniSpySDK/commit/da1d314e0e18fbd7c31c16747dede11dbc443946)
> [47230b3](https://github.com/GameProgressive/UniSpySDK/commit/47230b3c5d9993713f34748f1f0a1e8348f8a056)
> [2ff428e](https://github.com/GameProgressive/UniSpySDK/commit/2ff428ea6cd38f31b48c3a4d6f12f38718b7739d)
> [d3fcf36](https://github.com/GameProgressive/UniSpySDK/commit/d3fcf36a16de9134011bd234eb7d0a6a69b1638d)
> [3bef25d](https://github.com/GameProgressive/UniSpySDK/commit/3bef25df661ac64e108d213acdfcdd292d38b08a)
> [faff80e](https://github.com/GameProgressive/UniSpySDK/commit/faff80e6f3dd845325e77079a505bd8c550b7a82)
> [24ea6d4](https://github.com/GameProgressive/UniSpySDK/commit/24ea6d42ab5ca455a5f0a1e6aa144973e13e52cd)
> [5512fac](https://github.com/GameProgressive/UniSpySDK/commit/5512fac5d92b1043d62a07f483f60f48fde0d4da)
> [643ec30](https://github.com/GameProgressive/UniSpySDK/commit/643ec3000cdcc1fca21daa303127b8064f82c8c7)
> [fd27de0](https://github.com/GameProgressive/UniSpySDK/commit/fd27de0892c79a9c22dfe7e3392e1f6610365b8f)
> [e85e745](https://github.com/GameProgressive/UniSpySDK/commit/e85e7453f068572801e26d652f20abe18755ad34)
> 
> What if i split PR on three with roughly 25 commits per each? It will simplify review.

I think there's nothing else left based from the two conversations that were required to fix?

dimhotepus commented:
> > Thank you for patience!
> > Remaining commits to discuss:
> > https://github.com/GameProgressive/UniSpySDK/pull/27/files/173f1fe722b448fb33fa78a210abb25d0dd3d410..aacbb3061e11398dc5293f5d8ba38debd1f576f8#diff-35ca133c051513dc17a5b8f9280e7feefb421d0156883802222a85130a72c382
> > https://github.com/GameProgressive/UniSpySDK/pull/27/files/3078da53f97fda04deb6a8ceec4978e15e81e65f..3ff5871b6407d91b80fa85468cab847acef68fd4#diff-c3ba31498f33cf4a31e709e3d08cd8afca849076dc387a2b2f51e834e2655785
> > and you was interested in
> > [aacbb30](https://github.com/GameProgressive/UniSpySDK/commit/aacbb3061e11398dc5293f5d8ba38debd1f576f8)
> > [da1d314](https://github.com/GameProgressive/UniSpySDK/commit/da1d314e0e18fbd7c31c16747dede11dbc443946)
> > [47230b3](https://github.com/GameProgressive/UniSpySDK/commit/47230b3c5d9993713f34748f1f0a1e8348f8a056)
> > [2ff428e](https://github.com/GameProgressive/UniSpySDK/commit/2ff428ea6cd38f31b48c3a4d6f12f38718b7739d)
> > [d3fcf36](https://github.com/GameProgressive/UniSpySDK/commit/d3fcf36a16de9134011bd234eb7d0a6a69b1638d)
> > [3bef25d](https://github.com/GameProgressive/UniSpySDK/commit/3bef25df661ac64e108d213acdfcdd292d38b08a)
> > [faff80e](https://github.com/GameProgressive/UniSpySDK/commit/faff80e6f3dd845325e77079a505bd8c550b7a82)
> > [24ea6d4](https://github.com/GameProgressive/UniSpySDK/commit/24ea6d42ab5ca455a5f0a1e6aa144973e13e52cd)
> > [5512fac](https://github.com/GameProgressive/UniSpySDK/commit/5512fac5d92b1043d62a07f483f60f48fde0d4da)
> > [643ec30](https://github.com/GameProgressive/UniSpySDK/commit/643ec3000cdcc1fca21daa303127b8064f82c8c7)
> > [fd27de0](https://github.com/GameProgressive/UniSpySDK/commit/fd27de0892c79a9c22dfe7e3392e1f6610365b8f)
> > [e85e745](https://github.com/GameProgressive/UniSpySDK/commit/e85e7453f068572801e26d652f20abe18755ad34)
> > What if i split PR on three with roughly 25 commits per each? It will simplify review.
> 
> I think there's nothing else left based from the two conversations that were required to fix? Which I left a comment on both of them

Well, looks like all is fixed then and we can merge?

arves100 commented:
> 
> 
> > > Thank you for patience!
> > > Remaining commits to discuss:
> > > https://github.com/GameProgressive/UniSpySDK/pull/27/files/173f1fe722b448fb33fa78a210abb25d0dd3d410..aacbb3061e11398dc5293f5d8ba38debd1f576f8#diff-35ca133c051513dc17a5b8f9280e7feefb421d0156883802222a85130a72c382
> > > https://github.com/GameProgressive/UniSpySDK/pull/27/files/3078da53f97fda04deb6a8ceec4978e15e81e65f..3ff5871b6407d91b80fa85468cab847acef68fd4#diff-c3ba31498f33cf4a31e709e3d08cd8afca849076dc387a2b2f51e834e2655785
> > > and you was interested in
> > > [aacbb30](https://github.com/GameProgressive/UniSpySDK/commit/aacbb3061e11398dc5293f5d8ba38debd1f576f8)
> > > [da1d314](https://github.com/GameProgressive/UniSpySDK/commit/da1d314e0e18fbd7c31c16747dede11dbc443946)
> > > [47230b3](https://github.com/GameProgressive/UniSpySDK/commit/47230b3c5d9993713f34748f1f0a1e8348f8a056)
> > > [2ff428e](https://github.com/GameProgressive/UniSpySDK/commit/2ff428ea6cd38f31b48c3a4d6f12f38718b7739d)
> > > [d3fcf36](https://github.com/GameProgressive/UniSpySDK/commit/d3fcf36a16de9134011bd234eb7d0a6a69b1638d)
> > > [3bef25d](https://github.com/GameProgressive/UniSpySDK/commit/3bef25df661ac64e108d213acdfcdd292d38b08a)
> > > [faff80e](https://github.com/GameProgressive/UniSpySDK/commit/faff80e6f3dd845325e77079a505bd8c550b7a82)
> > > [24ea6d4](https://github.com/GameProgressive/UniSpySDK/commit/24ea6d42ab5ca455a5f0a1e6aa144973e13e52cd)
> > > [5512fac](https://github.com/GameProgressive/UniSpySDK/commit/5512fac5d92b1043d62a07f483f60f48fde0d4da)
> > > [643ec30](https://github.com/GameProgressive/UniSpySDK/commit/643ec3000cdcc1fca21daa303127b8064f82c8c7)
> > > [fd27de0](https://github.com/GameProgressive/UniSpySDK/commit/fd27de0892c79a9c22dfe7e3392e1f6610365b8f)
> > > [e85e745](https://github.com/GameProgressive/UniSpySDK/commit/e85e7453f068572801e26d652f20abe18755ad34)
> > > What if i split PR on three with roughly 25 commits per each? It will simplify review.
> > 
> > 
> > I think there's nothing else left based from the two conversations that were required to fix? Which I left a comment on both of them
> 
> Well, looks like all is fixed then and we can merge?

If there's anything else that you updated (expect for https://github.com/GameProgressive/UniSpySDK/pull/27/commits/e7916dda7fa0d5c3a4d31b4ef05b92b69af592ec which is ok) then yes I would merge it

dimhotepus commented:
> > > > Thank you for patience!
> > > > Remaining commits to discuss:
> > > > https://github.com/GameProgressive/UniSpySDK/pull/27/files/173f1fe722b448fb33fa78a210abb25d0dd3d410..aacbb3061e11398dc5293f5d8ba38debd1f576f8#diff-35ca133c051513dc17a5b8f9280e7feefb421d0156883802222a85130a72c382
> > > > https://github.com/GameProgressive/UniSpySDK/pull/27/files/3078da53f97fda04deb6a8ceec4978e15e81e65f..3ff5871b6407d91b80fa85468cab847acef68fd4#diff-c3ba31498f33cf4a31e709e3d08cd8afca849076dc387a2b2f51e834e2655785
> > > > and you was interested in
> > > > [aacbb30](https://github.com/GameProgressive/UniSpySDK/commit/aacbb3061e11398dc5293f5d8ba38debd1f576f8)
> > > > [da1d314](https://github.com/GameProgressive/UniSpySDK/commit/da1d314e0e18fbd7c31c16747dede11dbc443946)
> > > > [47230b3](https://github.com/GameProgressive/UniSpySDK/commit/47230b3c5d9993713f34748f1f0a1e8348f8a056)
> > > > [2ff428e](https://github.com/GameProgressive/UniSpySDK/commit/2ff428ea6cd38f31b48c3a4d6f12f38718b7739d)
> > > > [d3fcf36](https://github.com/GameProgressive/UniSpySDK/commit/d3fcf36a16de9134011bd234eb7d0a6a69b1638d)
> > > > [3bef25d](https://github.com/GameProgressive/UniSpySDK/commit/3bef25df661ac64e108d213acdfcdd292d38b08a)
> > > > [faff80e](https://github.com/GameProgressive/UniSpySDK/commit/faff80e6f3dd845325e77079a505bd8c550b7a82)
> > > > [24ea6d4](https://github.com/GameProgressive/UniSpySDK/commit/24ea6d42ab5ca455a5f0a1e6aa144973e13e52cd)
> > > > [5512fac](https://github.com/GameProgressive/UniSpySDK/commit/5512fac5d92b1043d62a07f483f60f48fde0d4da)
> > > > [643ec30](https://github.com/GameProgressive/UniSpySDK/commit/643ec3000cdcc1fca21daa303127b8064f82c8c7)
> > > > [fd27de0](https://github.com/GameProgressive/UniSpySDK/commit/fd27de0892c79a9c22dfe7e3392e1f6610365b8f)
> > > > [e85e745](https://github.com/GameProgressive/UniSpySDK/commit/e85e7453f068572801e26d652f20abe18755ad34)
> > > > What if i split PR on three with roughly 25 commits per each? It will simplify review.
> > > 
> > > 
> > > I think there's nothing else left based from the two conversations that were required to fix? Which I left a comment on both of them
> > 
> > 
> > Well, looks like all is fixed then and we can merge?
> 
> If there's not anything else that you updated (expect for [e7916dd](https://github.com/GameProgressive/UniSpySDK/commit/e7916dda7fa0d5c3a4d31b4ef05b92b69af592ec) which is ok) then yes I would merge it

Yep, nothing else.

arves100 commented:
Good, I have now merged all the changes! Thank you for your work during the initial analysis and reviews.
